### PR TITLE
Fix unit tests on darwin / non-linux platforms

### PR DESCRIPTION
**Pull request by smarterclayton**

### CODE REVIEW

**Comment by BenTheElder** [comment]
cc @spiffxp: [TestCRD flake again](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/55095/pull-kubernetes-unit/62708/)
/retest

File: pkg/kubectl/cmd/cp.go [hunk 1]
```diff
@@ -301,10 +301,12 @@ func recursiveTar(srcBase, srcFile, destBase, destFile string, tw *tar.Writer) e
...
 		}
-
 		defer f.Close()
-		_, err = io.Copy(tw, f)
-		return err
+
+		if _, err := io.Copy(tw, f); err != nil {
+			return err
+		}
+		return f.Close()
```
Comment (dixudx) [inline pkg/kubectl/cmd/cp.go]: Duplicate `f.Close()` in L304 and L309?

Remove L304.

Comment (smarterclayton) [reply pkg/kubectl/cmd/cp.go]: No, if we get an error we still need to close.  And you must close to guarantee the file is written to disk.

**Comment by spiffxp** [comment]
/lgtm

I can't `/approve no-issue` this, @smarterclayton you're welcome to if you'd like, or if there's an issue you can refer to, that'd make the bot happy

**Pull request merged**